packages/toolpad-app/src/components/MonacoEditor.tsx

```diff
@@ -183,6 +187,15 @@ export default React.forwardRef<MonacoEditorHandle, MonacoEditorProps>(function
       const pathUri = monaco.Uri.parse(`./scripts/${cuid()}${getExtension(language)}`);
       const model = monaco.editor.createModel(value || '', language, pathUri);
 
+      if (!overflowWidgetsDomNodeRef.current) {
+        const overflowWidgetsDomNode = document.createElement('div');
```

Comment:
NIT: just wondering - would it be possible to use `Portal` and just pass a ref? 🤔 
maybe something like this:
```jsx
<Portal>
  <div
    className='monaco-editor'
    ref={overflowWidgetsDomNodeRef}
    style={{ zIndex: overflowWidgetsDomNodeZindex }}
  />
</Portal>
```

Reply:
You mean using [`react-portal`](https://www.npmjs.com/package/react-portal)? I think that should be possible yes

Reply:
Hmm, I was thinking about one that MUI provides: https://mui.com/material-ui/react-portal/ but not sure if that's possible 🤔 anyways, that would apply if it doesn't complicate the matters, but seems that with MUI Portal we would need a way to pass container element all the way to body, unless we could do:
```jsx
<Portal container={document.body} />
```

Reply:
Just tried it out and it looks like the ref gets assigned later than the initial `useEffect` call, which means it's not available yet when the editor is created.

Looking at [the implementation](https://github.com/mui/material-ui/blob/1490632fc1d91a7d417e33f333db080293590301/packages/mui-base/src/Portal/Portal.js#L20) it makes sense, an effect has to run once before the portal gets rendered. I think this is done to avoid mismatches during hydration.